crates/external_services/src/grpc_client/dynamic_routing.rs

```diff
@@ -209,6 +215,24 @@ impl SuccessBasedDynamicRouting for SuccessRateCalculatorClient<Channel> {
 
         Ok(response)
     }
+
+    async fn invalidate_success_rate_routing_keys(
+        &self,
+        id: String,
+    ) -> DynamicRoutingResult<InvalidateWindowsResponse> {
+        let request = tonic::Request::new(InvalidateWindowsRequest { id });
+
+        let mut client = self.clone();
```

Comment:
Why are we cloning here?

Reply:
Cause we need a mutable ref to the Channel , and also cloning the channel is cheap